specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json

```diff
@@ -463,42 +471,69 @@
       ],
       "properties": {
         "query": {
-          "type": "string",
+          "type": "object",
           "description": "SAP ECC OData query. For example, \"$top=1\". Type: string (or Expression with resultType string)."
         }
       }
     },
-    "SapOpenHubSource": {
-      "description": "A copy activity source for SAP Business Warehouse Open Hub Destination source.",
+    "SqlSource": {
+      "description": "A copy activity SQL source.",
       "type": "object",
       "allOf": [
         {
           "$ref": "#/definitions/CopySource"
         }
       ],
-      "properties": {}
+      "properties": {
+        "sqlReaderQuery": {
+          "type": "object",
+          "description": "SQL reader query. Type: string (or Expression with resultType string)."
+        },
+        "sqlReaderStoredProcedureName": {
+          "type": "object",
+          "description": "Name of the stored procedure for a SQL Database source. This cannot be used at the same time as SqlReaderQuery. Type: string (or Expression with resultType string)."
+        },
+        "storedProcedureParameters": {
+          "description": "Value and type setting for stored procedure parameters. Example: \"{Parameter1: {value: \"1\", type: \"int\"}}\".",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/StoredProcedureParameter"
+          }
+        }
+      }
     },
-    "RestSource": {
```

Comment:
Did you remove this type? please make sure you are not deleting any of the existing types. Can you rearrange your changes so it is clear that you are only adding new things?

Reply:
Ok, I will submit another PR after solve these problems